source/shared/core_stmt.cpp

```diff
@@ -815,11 +815,11 @@ field_meta_data* core_sqlsrv_field_metadata( sqlsrv_stmt* stmt, SQLSMALLINT coln
     SQLLEN field_name_len = 0;
 
     meta_data = new ( sqlsrv_malloc( sizeof( field_meta_data ))) field_meta_data();
-    field_name_temp = static_cast<SQLWCHAR*>( sqlsrv_malloc( SS_MAXCOLNAMELEN * 2 + 1 ));
+    field_name_temp = static_cast<SQLWCHAR*>( sqlsrv_malloc( SS_MAXCOLNAMELEN * sizeof( SQLWCHAR ) + 1 ));
     SQLSRV_ENCODING encoding = ( (stmt->encoding() == SQLSRV_ENCODING_DEFAULT ) ? stmt->conn->encoding() :
         stmt->encoding());
 	try{
-        core::SQLDescribeColW( stmt, colno + 1, field_name_temp,  SS_MAXCOLNAMELEN * 2 + 1 , &field_len_temp,
+        core::SQLDescribeColW( stmt, colno + 1, field_name_temp,  (SS_MAXCOLNAMELEN + 1) * sizeof( SQLWCHAR ), &field_len_temp,
```

Comment:
And should the 4th parameter just be SS_MAXCOLNAMELEN since it is just character count?